### PR TITLE
Fix integration tests

### PR DESCRIPTION
**Pull request by vladciobanu**
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

File: src/main/haskell/kore/src/Kore/SMT/SMT.hs [hunk 1]
```diff
@@ -225,9 +230,11 @@ patternToSMT sloppy p =
             => CommonPurePattern Object
             -> String
             -> Translating (SBV a)
-        goLiteral (DV_ s (BuiltinDomainPattern (StringLiteral_ i))) hookName
+        goLiteral p@(DV_ s (BuiltinDomainPattern (StringLiteral_ i))) hookName
          | (getHookString $ getSortHook s) == hookName
-             = return $ literal $ read i
+             = liftEither $ note
+                 (MalformedDVLiteral p)
+                 (literal <$> readMaybe i)
```
Comment (ttuegel) [inline src/main/haskell/kore/src/Kore/SMT/SMT.hs]: The domain value literals should be parsed using the appropriate parser in `Kore.Builtin.Int` or `Kore.Builtin.Bool`. I think @u-- is taking care of this.

In the long-term, all these builtins will be decoded into an `Integer` or `Bool` before they ever get to SMT.

File: src/main/haskell/kore/src/Kore/Step/BaseStep.hs [hunk 2]
```diff
@@ -291,6 +291,18 @@ stepWithAxiom
...
         $ unificationProofStepVariablesToCommon
             existingVars variableMapping1 rawSubstitutionProof
+
+    let
+        variablesInLeftAxiom = pureAllVariables axiomLeftRaw
+        toVariable (AxiomVariable v) = v
+        toVariable (ConfigurationVariable v) = v
+        substitutions =
+            Set.fromList . fmap (toVariable . fst) $ normalizedSubstitution
+    if Predicate.isFalse condition
```
Comment (ttuegel) [inline src/main/haskell/kore/src/Kore/Step/BaseStep.hs]: The first part of this condition, `Predicate.isFalse condition`, does not seem to be related to the error message below; is that right?

File: src/main/haskell/kore/src/Kore/SMT/SMT.hs (hunk 1 above)
Comment (vladciobanu) [reply src/main/haskell/kore/src/Kore/SMT/SMT.hs]: The only reason for this change is to make the integration tests work again, for @traiansf 's PR. I would rather not do too big of a change, especially since @u-- has more changes coming up. This change was suggested by him, actually.
